msal/tests/Test.MSAL.NET.Unit/PublicClientApplicationTests.cs

```diff
@@ -550,35 +551,58 @@ public void GetUsersTest()
             MsalAccessTokenCacheItem item = new MsalAccessTokenCacheItem()
             {
                 Authority = TestConstants.AuthorityHomeTenant,
+                Environment = TestConstants.ProductionEnvironment,
                 ClientId = TestConstants.ClientId,
                 TokenType = "Bearer",
                 ExpiresOnUnixTimestamp =
                     CoreHelpers.DateTimeToUnixTimestamp((DateTime.UtcNow + TimeSpan.FromSeconds(3600))),
-                RawIdToken = MockHelpers.CreateIdToken(TestConstants.UniqueId, TestConstants.DisplayableId),
                 RawClientInfo = MockHelpers.CreateClientInfo(),
                 ScopeSet = TestConstants.Scope
             };
-            item.IdToken = IdToken.Parse(item.RawIdToken);
-            item.ClientInfo = ClientInfo.CreateFromJson(item.RawClientInfo);
-            item.AccessToken = item.GetAccessTokenItemKey().ToString();
+            item.CreateDerivedProperties();
```

Comment:
>CreateDerivedProperties [](start = 17, length = 23)

why is this needed? Can't the MsalAccessTokenCacheItem detect itself when that is needed (and e.g. ensure it does what ever it needs to do when someone calls a "calculated" property?

Reply:
it is possible, but since many properties (including key properties) are derived from idToken and client Info , CacheItem is almost useless without derived props

Reply:
imo: this complexity should be hidden inside the class. This code is too fragile imo

---
In reply to: [191035568](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1065#discussion_r191035568) [](ancestors = 191035568)